### PR TITLE
Add an option to set a transaction isolation level in start_transaction

### PR DESCRIPTION
**Pull request by volka**
This code lets a user select one of the four standardized transaction isolation levels when opening a transaction. Actual support dependes on the connectors, e.g. for SQLite only "serializable" and "read uncommitted" are meaningful.

I tried to implement the start_transaction() call in a way that should be backwards compatible with connectors not supporting the isolation_level parameter, but that makes for some possible confusion with the overloads of start_transaction.

For implementations see pull requests for the sqlite3 and postgresql connectors.

### CODE REVIEW

**Comment by volka** [comment]
Strange, Visual Studio does not seem to like "not" (though it supposedly should work with the /Za compiler flag). It is used quite often in the code ...

**Comment by rbock** [comment]
Hi Volker,

You need to add 

```
#include <ciso646>
```

Citing cppreference (http://en.cppreference.com/w/cpp/header/ciso646):

>  It may still be required in order to have the alternative operator representations in old or non-conformant compilers.

Best,

Roland

**Comment by rbock** [comment]
This is a nice addition.

I am not a big fan of default arguments. Maybe we could have two overloads (with/without isolation level)?

**Comment by volka** [comment]
Yes, I thought that might not be ideal, I'll fix that.

File: tests/MockDb.h [hunk 1]
```diff
@@ -244,6 +245,22 @@ struct MockDbT : public sqlpp::connection
   {
     return {name};
   }
+
+  void start_transaction(sqlpp::isolation_level level)
+  {
+    _current_isolation_level = level;
```
Comment (rbock) [inline tests/MockDb.h]: Storing the isolation level in the connection (even the mock) gives the wrong impression, I think.

**Comment by rbock** [comment]
Thanks!

File: tests/MockDb.h (hunk 1 above)
Comment (volka) [inline tests/MockDb.h]: Yes, that does not look nice, but I think this test is useful to verify that the correct internal start_transaction method gets called with the expected level.

I added an extra "InternalMockData" and moved the isolation_level there to make it explicit that this is only for testing.

Does this look better?

**Comment by rbock** [comment]
Actually, looking at the slite3 connector, I might have been wrong about this. If I read the documentation correctly, the isolation level is set per connection, not per transaction.

If that is correct, then the isolation level should indeed be a member of the connection (to be able to query it) and there should be a `get_isolation_level` and `set_isolation_level` method instead of just passing the parameter to `start_transaction`.

Or we need to reset the isolation level upon ending the transaction. But that might turn out pretty ugly...

**Pull request merged**